cli/lsp/repl.rs

```diff
@@ -0,0 +1,308 @@
+          process_id: None,
+          root_path: None,
+          root_uri: Some(cwd_uri.clone()),
+          initialization_options: Some(
+            serde_json::to_value(get_repl_workspace_settings()).unwrap(),
+          ),
+          capabilities: ClientCapabilities {
+            workspace: None,
+            text_document: None,
+            window: None,
+            general: None,
+            //offset_encoding: None,
+            experimental: None,
+          },
+          trace: None,
+          workspace_folders: None,
+          client_info: Some(ClientInfo {
+            name: "Deno REPL".to_string(),
+            version: None,
+          }),
+          locale: None,
+        })
+        .await?;
+
+      language_server.initialized(InitializedParams {}).await;
+
+      let server = ReplLanguageServer {
+        language_server,
+        document_version: 0,
+        document_text: String::new(),
+        pending_text: String::new(),
+        cwd_uri,
+      };
+      server.open_current_document().await;
+
+      Ok(server)
+    })
+    .await
+  }
+
+  pub async fn commit_text(&mut self, line_text: &str) {
+    with_logging_disabled(async {
+      self.did_change(line_text).await;
+      self.document_text.push_str(&self.pending_text);
+      self.pending_text = String::new();
+    })
+    .await
+  }
+
+  pub async fn completions(
+    &mut self,
+    line_text: &str,
+    position: usize,
+  ) -> Vec<ReplCompletionItem> {
+    with_logging_disabled(async {
+      self.did_change(line_text).await;
+      let before_line_len = BytePos(self.document_text.len() as u32);
+      let position = before_line_len + BytePos(position as u32);
+      let text_info = deno_ast::SourceTextInfo::from_string(format!(
+        "{}{}",
+        self.document_text, self.pending_text
+      ));
+      let line_and_column = text_info.line_and_column_index(position);
+      let response = self
+        .language_server
+        .completion(CompletionParams {
+          text_document_position: TextDocumentPositionParams {
+            text_document: TextDocumentIdentifier {
+              uri: self.get_document_specifier(),
+            },
+            position: Position {
+              line: line_and_column.line_index as u32,
+              character: line_and_column.column_index as u32,
+            },
+          },
+          work_done_progress_params: WorkDoneProgressParams {
+            work_done_token: None,
+          },
+          partial_result_params: PartialResultParams {
+            partial_result_token: None,
+          },
+          context: None,
+        })
+        .await
+        .ok()
+        .unwrap_or_default();
+
+      let items = match response {
+        Some(CompletionResponse::Array(items)) => items,
+        Some(CompletionResponse::List(list)) => list.items,
+        None => Vec::new(),
+      };
+      items
+        .into_iter()
+        .filter_map(|item| {
+          item.text_edit.and_then(|edit| match edit {
```

Comment:
In the future I believe we will be able to trivially improve this to return completions for more than just import specifier completions. Right now this somewhat limits it to just import specifier completions based on only handling this scenario.